spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkFilters.java

```diff
@@ -178,6 +182,10 @@ public static Expression convert(Filter filter) {
             // col NOT IN (1, 2) in Spark is equivalent to notNull(col) && notIn(col, 1, 2) in
             // Iceberg
             In childInFilter = (In) childFilter;
+            if (Stream.of(childInFilter.values()).anyMatch(Objects::isNull)) {
```

Comment:
Is it even realistic to get such filters past the optimizer? Won't they be replaced with false?

Reply:
That's one reason why we don't want to bother. If it has a NULL something is weird.

Reply:
If we want to be safer and not support such NOT IN predicates, I can see that. I am not yet convinced about IN predicates, however.